continuationsPlugin/src/it/scala/continuations/CompilerFixtures.scala

```diff
@@ -0,0 +1,896 @@
+ * bridge in sbt 0.13 [3], but has been heavily adapted and refactored. Original authors were
+ * Mark Harrah, Grzegorz Kossakowski, Martin Duhemm, Adriaan Moors and others.
+ *
+ * * dotty.tools.dotc.plugins: Adapted from scala/scala [1] with some modifications. They were
+ * originally authored by Lex Spoon, Som Snytt, Adriaan Moors, Paul Phillips and others.
+ *
+ * * dotty.tools.scaladoc: The Scaladoc documentation utility ships some third-party JavaScript
+ * and CSS libraries which are located under scaladoc/resources/dotty_res/styles/,
+ * scaladoc/resources/dotty_res/scripts/, docs/css/ and docs/js/. Please refer to the license
+ * header of the concerned files for details.
+ *
+ * * dotty.tools.dotc.coverage: Coverage instrumentation utilities have been adapted from the
+ * scoverage plugin for scala 2 [5], which is under the Apache 2.0 license.
+ *
+ * * The Dotty codebase contains parts which are derived from the ScalaPB protobuf library [4],
+ * which is under the Apache 2.0 license.
+ *
+ * [1] https://github.com/scala/scala
+ * [2] https://github.com/adriaanm/scala/tree/sbt-api-consolidate/src/compiler/scala/tools/sbt
+ * [3] https://github.com/sbt/sbt/tree/0.13/compile/interface/src/main/scala/xsbt
+ * [4] https://github.com/lampepfl/dotty/pull/5783/files
+ * [5] https://github.com/scoverage/scalac-scoverage-plugin
+ * 
+ * The below fixtures and methods were adapted from the dotty tools DottyTest class [6].
+ * https://github.com/lampepfl/dotty/blob/main/compiler/test/dotty/tools/DottyTest.scala
+ *   - Accessed on 2022/10/08T22:20:00.000Z-5:00 [6]
+ */
+// format: on
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.ast.Trees
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.core.Comments.ContextDoc
+import dotty.tools.dotc.core.Comments.ContextDocstrings
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Contexts.ContextBase
+import dotty.tools.dotc.core.Contexts.ctx
+import dotty.tools.dotc.core.Decorators.*
+import dotty.tools.dotc.core.Definitions
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.core.Flags.EmptyFlags
+import dotty.tools.dotc.core.Names
+import dotty.tools.dotc.core.Phases.Phase
+import dotty.tools.dotc.core.StdNames.nme
+import dotty.tools.dotc.core.Symbols
+import dotty.tools.dotc.core.Symbols.*
+import dotty.tools.dotc.core.Types
+import dotty.tools.dotc.core.Types.ContextualMethodType
+import dotty.tools.dotc.core.Types.MethodType
+import dotty.tools.dotc.core.Types.Type
+import dotty.tools.dotc.core.Types.TypeRef
+import munit.FunSuite
+
+import scala.concurrent.Promise
+import scala.util.Properties
+import scala.util.Try
+
+trait CompilerFixtures { self: FunSuite =>
+  def continuationsCompilerSnapshot(source: String)(using Context): (String, String) =
+    val p = Promise[(String, String)]
+    checkContinuations(source)((t, c) => p.complete(Try((t.show, c.toString))))
```

Comment:
I am not quite sure what having the context here gives us, as you can that the information in the file doesn't show anything useful.
also it includes things like `MutableScope@163a9512` which is an instance that has a different name on every run so actually it can't be used. I tried to use the `c.scope.toList` but that comes back empty